docs/source/concepts/lookup.rst

```diff
@@ -142,16 +142,201 @@ Female  40         60       30
 
+Constructing Lookup Tables from a Component
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Components can register lookup tables to be built by specifying
+a ``data_sources`` block in their :attr:`~vivarium.component.Component.configuration_defaults` property. 
+As a basic example, DiseaseModel in ``vivarium_public_health`` has the following
+``data_sources`` configuration:
+
+.. code-block:: python
+
+    @property
+    def configuration_defaults(self) -> dict[str, Any]:
+        return {
+            f"{self.name}": {
+                "data_sources": {
+                    "cause_specific_mortality_rate": self.load_cause_specific_mortality_rate,
+                },
+            },
+        }
+
+which specifies a single lookup table named
+``cause_specific_mortality_rate`` whose data is provided by the component's
+``load_cause_specific_mortality_rate`` method.
+
+Each entry in
+``data_sources`` maps a table name to a data source from one of several supported types
+(see `Data Source Types`_). Barring edge cases (see
+`Limitations and When to Override`_), one should specify all of a component's
+lookup tables via ``data_sources``, instead of accessing the builder's lookup
+interface directly.
+
+When a component configures ``data_sources``, the base
+:class:`Component <vivarium.component.Component>` class automatically builds
+the lookup tables before the component's :meth:`~vivarium.component.Component.setup` method is called. The
+resulting tables are stored in the component's :attr:`~vivarium.component.Component.lookup_tables` dictionary,
+keyed by the name specified in ``data_sources``. 
+
+This approach separates the *what* (which tables to build and where to get data) from the
+*how* (the mechanics of table construction), making components easier to
+write and configure. It also allows users to override data sources in model specification files
+without modifying component code. Following the example above, a model specification could adjust the 
+``cause_specific_mortality_rate`` data source to point to different data or a scalar value:
+
+.. code-block:: yaml
+
+    configuration:
+        disease_model:
+            data_sources:
+                cause_specific_mortality_rate: 0.02
+
+Data Source Types
+^^^^^^^^^^^^^^^^^
+
+Each entry in ``data_sources`` maps a table name to a data source. The
+following data source types are supported:
+
+**Artifact key (string without** ``::`` **):**
+    A string path to data in the artifact, e.g.,
+    ``"cause.all_causes.cause_specific_mortality_rate"``. The data is loaded
+    via :meth:`builder.data.load() <vivarium.framework.artifact.interface.ArtifactInterface.load>`. Strings with ``::`` are reserved for method
+    or function references (see below).
+
+**Callable:**
+    Any callable (function, lambda, or bound method) that accepts a ``builder``
+    argument and returns the data.
+
+**Scalar value:**
+    A numeric value (``int``, ``float``), ``datetime``, or ``timedelta`` that
```

Comment:
I think there is a Numberlike object in types.py but I'm not sure if it is better to use the more common data types (what you have here) or the special data types we have created in that module.